node/delivery/deliver_client.go

```diff
@@ -63,18 +63,23 @@ func SingleSpecifiedSeekInfo(seq uint64) *orderer.SeekInfo {
 
 func Pull(context context.Context, channel string, logger types.Logger, endpoint func() string, requestEnvelopeFactory func() *common.Envelope, cc comm.ClientConfig, handleBlock func(block *common.Block), onClose func()) {
 	logger.Infof("Started pulling from: %s", channel)
-	for {
-		time.Sleep(time.Second)
 
-		select {
-		case <-context.Done():
-			logger.Infof("Returning since context is done")
-			if onClose != nil {
-				onClose()
-			}
-			return
-		default:
+	time.Sleep(time.Second)
```

Comment:
why sleep for 1s before even starting? I think the first attempt should be done immediately